<launch>
  <arg name="stereo" default="stereo"/>
  <arg name="left" default="left"/>
  <arg name="right" default="right"/>
  <arg name="image" default="image_rect"/>

  <arg name="pub_tf" default="false"/>
  <arg name="max_features" default="2"/>
  <arg name="vis_matches" default="true"/>

  <arg name="output" default="screen"/>

  <node pkg="viso2_ros" type="stereo_odometer" name="stereo_odometer"
    output="$(arg output)" ns="$(arg stereo)">
    <param name="odom_frame_id" type="string" value="world"/>
    <param name="max_features" type="int" value="$(arg max_features)"/>
    <param name="pub_tf" type="bool" value="$(arg pub_tf)"/>
    <param name="vis_matches" type="bool" value="$(arg vis_matches)"/>
    <param name="pub_pose" type="bool" value="true"/>
    <param name="pub_odom" type="bool" value="true"/>

    <remap from="~odometry" to="odom"/>
    <remap from="~pose" to="pose"/>
    <remap from="~left" to="$(arg left)"/>
    <remap from="~right" to="$(arg right)"/>
    <remap from="~left/image" to="$(arg left)/$(arg image)"/>
    <remap from="~right/image" to="$(arg right)/$(arg image)"/>
  </node>
</launch>
